app/models/competition.rb

```diff
@@ -2883,4 +2899,9 @@ def self.form_data_json_schema
       },
     }
   end
+
+  def auto_close
```

Comment:
This only attempts to auto-close, so the name should be `attempt_auto_close`. With that in mind, you should return a boolean to indicate whether it auto-closed or not. That could be as simple as `threshold_reached && update!(...)` because the latter `update` call also returns a boolean in Rails if the update was successful.
Lastly, it does modify the competition in-place so the method name should end with a `!` per Ruby conventions.

Reply:
Changes made